concent_api/core/views.py

```diff
@@ -1225,6 +1245,219 @@ def store_message_and_message_status(
         receive_message_status.save()
 
 
+def store_subtask(
+    task_id:                        str,
+    subtask_id:                     str,
+    provider_public_key:            str,
+    requestor_public_key:           str,
+    state:                          str,
+    next_deadline:                  datetime.datetime,
+    task_to_compute:                message.TaskToCompute,
+    report_computed_task:           message.ReportComputedTask,
+    ack_report_computed_task:       message.AckReportComputedTask,
+    reject_report_computed_task:    message.RejectReportComputedTask,
+    subtask_results_accepted:       message.tasks.SubtaskResultsAccepted,
+    subtask_results_rejected:       message.tasks.SubtaskResultsRejected,
+):
+
+    provider  = Client.objects.get_or_create(provider_public_key)
+    requestor = Client.objects.get_or_create(requestor_public_key)
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state,
+        next_deadline   = next_deadline,
+    )
+
+    if task_to_compute is not None:
+        stored_message = store_message(
+            task_to_compute,
+            task_id,
+            subtask_id,
+        )
+        subtask.task_to_compute = stored_message
+
+    if report_computed_task is not None:
+        stored_message = store_message(
+            report_computed_task,
+            task_id,
+            subtask_id,
+        )
+        subtask.report_computed_task = stored_message
+
+    if ack_report_computed_task is not None:
+        stored_message = store_message(
+            ack_report_computed_task,
+            task_id,
+            subtask_id,
+        )
+        subtask.ack_report_computed_task = stored_message
+
+    if reject_report_computed_task is not None:
+        stored_message = store_message(
+            reject_report_computed_task,
+            task_id,
+            subtask_id,
+        )
+        subtask.reject_report_computed_task = stored_message
+
+    if subtask_results_accepted is not None:
+        stored_message = store_message(
+            subtask_results_accepted,
+            task_id,
+            subtask_id,
+        )
+        subtask.subtask_results_accepted = stored_message
+
+    if subtask_results_rejected is not None:
+        stored_message = store_message(
+            subtask_results_rejected,
+            task_id,
+            subtask_id,
+        )
+        subtask.subtask_results_rejected = stored_message
+
+    subtask.full_clean()
+    subtask.save()
+
+
+def store_specific_message(
```

Comment:
This should be called just `store_message()`. Or `store_typed_message()`.

Please avoid using the word "specific" in names. It's another word that - like "check" - is just filler that says nothing specific :)